In [1]:
import pandas as pd
import seaborn as sns

In [2]:
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import style

In [3]:
df=pd.read_csv(r'C:\atlantis_citizens_final.csv')
pd.set_option('display.max_columns',10)

In [4]:
df

Citizen_ID       Diet_Type    District_Name Occupation  Wealth_Index  \
0      CIT_15935  Exotic Imports      Coral Slums     Scribe        1491.0   
1      CIT_11623         Seafood      Coral Slums     Fisher        1596.0   
2       CIT_8026         Seafood    Mariana Plaza    Warrior        3921.0   
3       CIT_0492  Exotic Imports      Deep Trench     Fisher           NaN   
4       CIT_0275         Seaweed      Deep Trench    Warrior       25985.0   
...          ...             ...              ...        ...           ...   
15746  CIT_10602         Seaweed    Mariana Plaza   Merchant        2896.0   
15747   CIT_0865         Seaweed      Coral Slums      Miner        1671.0   
15748   CIT_0809  Exotic Imports    Mariana Plaza     Scribe        4656.0   
15749  CIT_10750         Seafood  The Golden Reef   Merchant       17529.0   
15750   CIT_6381         Seaweed      Coral Slums    Warrior        1439.0   

       House_Size_sq_ft  Life_Expectancy Vehicle_Owned    Work_District  \
0                 100.0             42.0   Fin Bicycle    Mariana Plaza   
1                 100.0             49.0   Sea Scooter      Deep Trench   
2                 533.0             37.0   Sea Scooter      Deep Trench   
3                 136.0             38.0   Fin Bicycle      Deep Trench   
4                2673.0             54.0   Sea Scooter      Deep Trench   
...                 ...              ...           ...              ...   
15746             428.0             59.0   Sea Scooter    Mariana Plaza   
15747             100.0             47.0    No Vehicle    Mariana Plaza   
15748             756.0             53.0   Fin Bicycle    Mariana Plaza   
15749            3961.0             72.0    No Vehicle  The Golden Reef   
15750             100.0             32.0   Sea Scooter      Coral Slums   

       Bio_Hash  
0      b81cb8ce  
1      72f48eef  
2      0abde296  
3      8055fc9e  
4      c77829e2  
...         ...  
15746  268659a4  
15747  ca287a73  
15748  717ca395  
15749  34fb8c73  
15750  ac857e92  

[15751 rows x 10 columns]

In [5]:
df.head(10)

Citizen_ID       Diet_Type    District_Name Occupation  Wealth_Index  \
0  CIT_15935  Exotic Imports      Coral Slums     Scribe        1491.0   
1  CIT_11623         Seafood      Coral Slums     Fisher        1596.0   
2   CIT_8026         Seafood    Mariana Plaza    Warrior        3921.0   
3   CIT_0492  Exotic Imports      Deep Trench     Fisher           NaN   
4   CIT_0275         Seaweed      Deep Trench    Warrior       25985.0   
5   CIT_7475         Seaweed    Mariana Plaza    Warrior        6173.0   
6  CIT_17545  Exotic Imports      Coral Slums     Scribe        1506.0   
7   CIT_5234  Exotic Imports  The Golden Reef    Warrior       19054.0   
8  CIT_10281         Seafood      Coral Slums      Miner        1890.0   
9  CIT_14760         Seafood  The Golden Reef      Miner        2597.0   

   House_Size_sq_ft  Life_Expectancy Vehicle_Owned  Work_District  Bio_Hash  
0             100.0             42.0   Fin Bicycle  Mariana Plaza  b81cb8ce  
1             100.0             49.0   Sea Scooter    Deep Trench  72f48eef  
2             533.0             37.0   Sea Scooter    Deep Trench  0abde296  
3             136.0             38.0   Fin Bicycle    Deep Trench  8055fc9e  
4            2673.0             54.0   Sea Scooter    Deep Trench  c77829e2  
5             911.0             47.0   Sea Scooter    Deep Trench  e4023d89  
6             100.0             51.0    No Vehicle  Mariana Plaza  6eabb698  
7            4314.0             47.0     Submarine    Coral Slums  ada3f024  
8               NaN             45.0   Sea Scooter    Coral Slums  7b32810b  
9               NaN             43.0   Fin Bicycle    Coral Slums  addf2367

In [6]:
df.shape

(15751, 10)

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15751 entries, 0 to 15750
Data columns (total 10 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Citizen_ID        15751 non-null  object 
 1   Diet_Type         15751 non-null  object 
 2   District_Name     15751 non-null  object 
 3   Occupation        15751 non-null  object 
 4   Wealth_Index      14696 non-null  float64
 5   House_Size_sq_ft  14554 non-null  float64
 6   Life_Expectancy   15137 non-null  float64
 7   Vehicle_Owned     15751 non-null  object 
 8   Work_District     15751 non-null  object 
 9   Bio_Hash          15751 non-null  object 
dtypes: float64(3), object(7)
memory usage: 1.2+ MB







Observations
There are a total of 15751 rows and 10 columns in data set
Dataset contains both numerical figures as well as categorical features
Different columns represent different data as demographics,lifestyle and economic indicators

In [8]:
df.columns

Index(['Citizen_ID', 'Diet_Type', 'District_Name', 'Occupation',
       'Wealth_Index', 'House_Size_sq_ft', 'Life_Expectancy', 'Vehicle_Owned',
       'Work_District', 'Bio_Hash'],
      dtype='object')

In [9]:
df.dtypes

Citizen_ID           object
Diet_Type            object
District_Name        object
Occupation           object
Wealth_Index        float64
House_Size_sq_ft    float64
Life_Expectancy     float64
Vehicle_Owned        object
Work_District        object
Bio_Hash             object
dtype: object


some columns have float value and some have string variables

In [10]:
df['Wealth_Index']

0         1491.0
1         1596.0
2         3921.0
3            NaN
4        25985.0
          ...   
15746     2896.0
15747     1671.0
15748     4656.0
15749    17529.0
15750     1439.0
Name: Wealth_Index, Length: 15751, dtype: float64

In [11]:
df.isna().sum()

Citizen_ID             0
Diet_Type              0
District_Name          0
Occupation             0
Wealth_Index        1055
House_Size_sq_ft    1197
Life_Expectancy      614
Vehicle_Owned          0
Work_District          0
Bio_Hash               0
dtype: int64

In [12]:
df.loc[df.duplicated()]

Empty DataFrame
Columns: [Citizen_ID, Diet_Type, District_Name, Occupation, Wealth_Index, House_Size_sq_ft, Life_Expectancy, Vehicle_Owned, Work_District, Bio_Hash]
Index: []



none of the data is duplicated

In [13]:
df.loc[df.duplicated(subset=['Occupation'])]

Citizen_ID       Diet_Type    District_Name Occupation  Wealth_Index  \
3       CIT_0492  Exotic Imports      Deep Trench     Fisher           NaN   
4       CIT_0275         Seaweed      Deep Trench    Warrior       25985.0   
5       CIT_7475         Seaweed    Mariana Plaza    Warrior        6173.0   
6      CIT_17545  Exotic Imports      Coral Slums     Scribe        1506.0   
7       CIT_5234  Exotic Imports  The Golden Reef    Warrior       19054.0   
...          ...             ...              ...        ...           ...   
15746  CIT_10602         Seaweed    Mariana Plaza   Merchant        2896.0   
15747   CIT_0865         Seaweed      Coral Slums      Miner        1671.0   
15748   CIT_0809  Exotic Imports    Mariana Plaza     Scribe        4656.0   
15749  CIT_10750         Seafood  The Golden Reef   Merchant       17529.0   
15750   CIT_6381         Seaweed      Coral Slums    Warrior        1439.0   

       House_Size_sq_ft  Life_Expectancy Vehicle_Owned    Work_District  \
3                 136.0             38.0   Fin Bicycle      Deep Trench   
4                2673.0             54.0   Sea Scooter      Deep Trench   
5                 911.0             47.0   Sea Scooter      Deep Trench   
6                 100.0             51.0    No Vehicle    Mariana Plaza   
7                4314.0             47.0     Submarine      Coral Slums   
...                 ...              ...           ...              ...   
15746             428.0             59.0   Sea Scooter    Mariana Plaza   
15747             100.0             47.0    No Vehicle    Mariana Plaza   
15748             756.0             53.0   Fin Bicycle    Mariana Plaza   
15749            3961.0             72.0    No Vehicle  The Golden Reef   
15750             100.0             32.0   Sea Scooter      Coral Slums   

       Bio_Hash  
3      8055fc9e  
4      c77829e2  
5      e4023d89  
6      6eabb698  
7      ada3f024  
...         ...  
15746  268659a4  
15747  ca287a73  
15748  717ca395  
15749  34fb8c73  
15750  ac857e92  

[15746 rows x 10 columns]

There are duplicate values in occupatuion type suggesting common occupations also many other columns are duplicated which is quite reasonable as lifetyle are kind of common among people

Explaining the name of the columns
Citizen Id-gives uniqueness to each citizen
Diet Type-Kind of food comsumed by the citizen
District Name-District the citizen lives in
Occupation-the work he does to make living
Wealth Index-Kind of Net worth
House Size-Self Explanatory tells about the house size
Life Expectancy-How much years does a person live
Vehicles Owned-what vehicles do these people own
Work District-The place where they work
bio hash-no idea what they are

In [ ]:
df['House_Size_sq_ft'].value_counts()


Initial Hypothesis
People with more income will tend to have big houses
life expectancy may increase with income 
occupation is directly proportional to income

In [ ]:
district_wealth = (
    df.groupby('District_Name')['Wealth_Index']
      .agg(['mean', 'median'])
      .sort_values('mean', ascending=False)
)

district_wealth

In [ ]:
style.use("dark_background")
district_wealth.plot(
    kind='bar',
    figsize=(10,5),
)

plt.ylabel('Wealth')
plt.title('Average and Median Wealth by District')
plt.xticks(rotation=45)
plt.show()

Q1-Compare district wise wealth index
Ans-The golden reef has the highest weakth while coral slums has the lowest wealth index
In all the districts we can see that kean wealth is higher than median wealth
the gap between mean and median shows the wealth inequality
Median income provides better comparative measure than mean income

In [ ]:
district_health = (
    df.groupby('District_Name')
      .agg(
          avg_wealth_index=('Wealth_Index', 'mean'),
          avg_life_expectancy=('Life_Expectancy', 'mean')
      )
)

district_health

In [ ]:
plt.figure(figsize=(7,5))
plt.plot(
    district_health['avg_wealth_index'],
    district_health['avg_life_expectancy'],c="red"
)

plt.xlabel('Average Wealth Index by District')
plt.ylabel('Average Life Expectancy by District')
plt.title('District Wealth Index vs Life Expectancy')
plt.show()

Ans-Sadly its the reality as the visual and the numerical data suggests
reason to this might be the healthcare facilities available to the people because of higher income
in graph a slight decrease is seen which can be treated as a rare case and that decline also happens in lower wealth region so kind of outlier
thi decrease shows that life expectancy is not full dependent on the wealth

In [ ]:
occupation_income = (
    df.groupby('Occupation')['Wealth_Index']
      .agg(['mean', 'median'])
      .sort_values('median', ascending=False)
)

occupation_income

In [ ]:
occupation_income.plot(kind='bar',figsize=(10,5))
plt.ylabel('Income')
plt.title('Mean and Median Wealth Index by Occupation')
plt.xticks(rotation=45)
plt.show()

Q3-does occupatipn have any correltion with wealth
ANS-straight up the answer is yes for sure it does.
the plot clearly shows that merchant have higher wealth index which kind of leads to income inequlity as the gap between the merchant and the others is  huge
simple logic why merchants are paid higher is because of better skills


In [ ]:
df[['Wealth_Index', 'House_Size_sq_ft']].describe()

In [ ]:
plt.figure(figsize=(7,5))
plt.scatter(
    df['Wealth_Index'],
    df['House_Size_sq_ft'],
    c="blue",alpha=0.4
)

plt.xlabel('wealth')
plt.ylabel('House Size')
plt.title('Wealth vs House Size')
plt.show()

Q4-Does house size dedpend on incone?
Ans-as expected yes since more the income more can the person invest in housing and improving the lifetyle.
    but at higher wealth levels rate of increase of house size decreases as there are constraints such as family needs and location

In [ ]:
df[['House_Size_sq_ft','Life_Expectancy']].describe()

In [ ]:
plt.figure(figsize=(7,5))
plt.scatter(df['House_Size_sq_ft'],df['Life_Expectancy'],alpha=0.3)
plt.xlabel('House_Size_sq_ft')
plt.ylabel('Life_Expectancy')
plt.title('House Size vs Life Expectancy')
plt.show()

Q5-is there any relationship between life expectancy and house size?
Ans-this was quite difficult to analyze and seems quite non trivial.
Life expectancy shows a slight increase with larger house sizes
the relationship though shatters as we go up in bigger houses
large houses reflect better housing,less overcrowding and better sanitation
however house size does not alone capture healthcare

In [ ]:
plt.figure(figsize=(8, 6))
sns.heatmap(
    cm,
    annot=True,
    fmt="d",
    cmap="Blues",
    xticklabels=sorted(y.unique()),
    yticklabels=sorted(y.unique())
)
plt.xlabel("Predicted Occupation")
plt.ylabel("Actual Occupation")
plt.title("Confusion Matrix: Occupation Classification")
plt.show()

In [ ]:
cm_df = pd.DataFrame(cm, index=sorted(y.unique()), columns=sorted(y.unique()))

confused_pairs = (
    cm_df.where(~np.eye(cm_df.shape[0], dtype=bool))
         .stack()
         .sort_values(ascending=False))

confused_pairs.head(5)

Q6-Are there occupations that are systematically confused with each other?